src/main/java/Duke.java

```diff
@@ -1,10 +1,64 @@
+import Exceptions.DukeException;
+import java.util.Scanner;
 public class Duke {
+    private final Ui _ui;
+    private TaskList _taskList;
```

Comment:
Perhaps you should name your variable as taskList instead of _taskList. I think it's better not to let a variable name start with a hyphen. It's good practice to use camelCase for your variable names.